components/inputs/input-date.js

```diff
@@ -155,25 +159,35 @@ class InputDate extends LocalizeStaticMixin(LitElement) {
 		this._dropdown = this.shadowRoot.querySelector('d2l-dropdown-content');
 		this._calendar = this.shadowRoot.querySelector('d2l-calendar');
 
+		this.addEventListener('blur', this._handleBlur);
 		this.addEventListener('d2l-localize-behavior-language-changed', () => {
 			this._dateTimeDescriptor = getDateTimeDescriptorShared(true);
-			this.requestUpdate();
+			this._getContentWidth();
 		});
+
+		this._formattedValue = this.emptyText ? this.emptyText : '';
+		const fontsPromise = document.fonts ? document.fonts.ready : Promise.resolve();
+		await Promise.resolve(fontsPromise);
+		this._getContentWidth();
 	}
 
 	render() {
+		const shortDateFormat = (this._dateTimeDescriptor.formats.dateFormats.short).toUpperCase();
+		const inputTextWidth = this._contentWidth ? `calc(${this._contentWidth}px + 1.2rem + 0.75rem + 2px)` : '9rem'; // text and icon width + icon margin left and right + paddingRight + border width
 		return html`
 			<d2l-dropdown ?disabled="${this.disabled}" no-auto-open>
 				<d2l-input-text
 					@change="${this._handleChange}"
 					class="d2l-dropdown-opener"
 					?disabled="${this.disabled}"
+					@focus="${this._handleInputTextFocus}"
 					@keydown="${this._handleKeydown}"
 					label="${ifDefined(this.label)}"
 					?label-hidden="${this.labelHidden}"
 					@mouseup="${this._handleMouseup}"
-					placeholder="${(this._dateTimeDescriptor.formats.dateFormats.short).toUpperCase()}"
-					title="${this.localize('openInstructions')}"
+					placeholder="${shortDateFormat}"
+					style="${styleMap({maxWidth: inputTextWidth})}"
```

Comment:
With `max-width` set, what's actually causing it to become wide? I guess it's the text input's `width: 100%`?